server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -837,20 +856,35 @@ private long processVirtLimitPools(List<Pool> existingPools,
                             // adjustment for exported quantities if there are multiple
                             // pools in play.
                             long adjust = 0L;
+                            boolean isMasterUnlimited = false;
+
                             for (Pool derivedPool : existingPools) {
                                 String isDerived =
                                     derivedPool.getAttributeValue(Pool.Attributes.DERIVED_POOL);
 
                                 if (isDerived == null) {
                                     adjust = derivedPool.getExported();
+
+                                    if (derivedPool.getQuantity() == UNLIMITED_QUANTITY) {
+                                        isMasterUnlimited = true;
```

Comment:
we also need to make sure the `existingPool` whose quantity we're trying to calculate is either a BONUS or UNMAPPED_GUEST type, and only then set expectedQuantity = -1

Reply:
Changes made. This check is also added in EntitlementRules [postUnbindVirtLimit](https://github.com/candlepin/candlepin/blob/aade17f8fa0e6db5a64c0f2a439a7f9ec1c22c70/server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRules.java#L546) & [adjustHostedBonusPoolQuantity](https://github.com/candlepin/candlepin/blob/aade17f8fa0e6db5a64c0f2a439a7f9ec1c22c70/server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRules.java#L733)